presentation-compiler/src/main/dotty/tools/pc/Worker.scala

```diff
@@ -0,0 +1,107 @@
+package dotty.tools.pc
+
+import java.util.concurrent.ScheduledExecutorService
+import scala.meta.pc.PresentationCompilerConfig
+import scala.meta.internal.metals.Report
+import java.util.logging.Logger
+import java.util.logging.Level
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.ScheduledFuture
+
+case class InfiniteCompilationException(msg: String) extends Exception // we should expose this in mtags-interfaces so we can catch it in IDE's
+
+class Worker(queue: TaskQueue, config: PresentationCompilerConfig, reportContext: PcReportContext):
+  private val logger: Logger = Logger.getLogger(getClass.getName).nn
+
+  private val executor: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor().nn
+  private val timeoutExecutor: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor().nn
+  private var _currentTask: AtomicReference[Option[queue.Task[?]]] = AtomicReference(None)
+
+  def scheduleProcessing(): Unit =
+    executor.execute(() => processQueue())
```

Comment:
So this is called in `enqueueTask` and it calls `processQueue`, but `processQueue` also calls itself. Won't that queue a lot of `processQueue` calls?